x-pack/plugins/security/server/authentication/authenticator.test.ts

```diff
@@ -0,0 +1,654 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+jest.mock('./providers/basic', () => ({ BasicAuthenticationProvider: jest.fn() }));
+
+import Boom from 'boom';
+import { SessionStorage } from '../../../../../src/core/server';
+
+import {
+  loggingServiceMock,
+  httpServiceMock,
+  httpServerMock,
+  elasticsearchServiceMock,
+  sessionStorageMock,
+} from '../../../../../src/core/server/mocks';
+import { mockAuthenticatedUser } from '../../common/model/authenticated_user.mock';
+import { AuthenticationResult } from './authentication_result';
+import { Authenticator, AuthenticatorOptions, ProviderSession } from './authenticator';
+import { DeauthenticationResult } from './deauthentication_result';
+import { BasicAuthenticationProvider } from './providers';
+
+function getMockOptions(config: Partial<AuthenticatorOptions['config']> = {}) {
+  return {
+    clusterClient: elasticsearchServiceMock.createClusterClient(),
+    basePath: httpServiceMock.createSetupContract().basePath,
+    loggers: loggingServiceMock.create(),
+    isSystemAPIRequest: jest.fn(),
+    getServerBaseURL: jest.fn(),
```

Comment:
__note:__ diff#4: instead of passing protocol/hostname/port tripple I just pass a function that generates server base URL. It's required since we get necessary info from the legacy plugin and it's actually shows the intention better.